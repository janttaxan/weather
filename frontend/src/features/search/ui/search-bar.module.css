@import "shared/variables.css";

.root {}

.input {}
