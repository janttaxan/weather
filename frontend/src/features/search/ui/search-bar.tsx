import styles from './search-bar.module.css';

import classNames from 'classnames';

interface SearchBarProps {
  className?: string;
}

export const SearchBar = (props: SearchBarProps) => {
  return (
    <div className={classNames(styles.root, props.className)}>
      <input className={styles.input} type='text' placeholder='Поиск' />
    </div>
  );
};
