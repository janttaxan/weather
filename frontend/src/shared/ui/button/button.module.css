.root {
  background-color: transparent;
  cursor: pointer;
}
