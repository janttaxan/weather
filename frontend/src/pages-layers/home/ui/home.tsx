import { Button } from 'shared/ui/button';
import { Page } from 'widgets/page';

export const Home = () => {
  return (
    <Page context={{ user: null }}>
      <div>
        init project
        <Button text='button' />
      </div>
    </Page>
  );
};
