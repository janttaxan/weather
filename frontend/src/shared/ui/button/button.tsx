import styles from './button.module.css';

import classNames from 'classnames';

import { ButtonProps } from 'shared/ui/button/types';

export const Button = (props: ButtonProps) => {
  return (
    <button className={classNames(styles.root, props.className)} onClick={props.onClick} data-testid='button'>
      {props.text}
    </button>
  );
};
