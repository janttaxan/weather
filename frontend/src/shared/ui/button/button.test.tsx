import { render, getByTestId } from '@testing-library/react';

import { Button } from 'shared/ui/button';

describe('Button component', () => {
  it('render a button', () => {
    render(<Button text='test button' />);
    expect(getByTestId(document.documentElement, 'button')).toBeInTheDocument();
  });

  it('renders button unchanged', () => {
    const { container } = render(<Button text='test' />);
    expect(container).toMatchSnapshot();
  });
});
