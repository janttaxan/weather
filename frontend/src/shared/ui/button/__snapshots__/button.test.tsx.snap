// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders button unchanged 1`] = `
<div>
  <button
    class="root"
    data-testid="button"
  >
    test
  </button>
</div>
`;
